src/graphql/data/TopTokens.ts

```diff
@@ -132,18 +132,18 @@ function toContractInput(token: PrefetchedTopToken) {
   }
 }
 
-// Map of key: ${chain} + ${address} and value: TopToken object.
+// Map of key: ${HistoryDuration} and value: another Map, of key:${chain} + ${address} and value: TopToken object.
 // Acts as a local cache.
-const tokensWithPriceHistoryCache: Record<string, TopToken> = {}
+const tokensWithPriceHistoryCache: Record<string, Record<string, TopToken>> = {}
 
-const checkIfAllTokensCached = (tokens: PrefetchedTopToken[]) => {
+const checkIfAllTokensCached = (duration: string, tokens: PrefetchedTopToken[]) => {
```

Comment:
iirc there is a HistoryDuration type somewhere that we could use here

Reply:
yeah although typescript  forces us to predefine the key values and include '%future added value'. I guess this works fine though? (changed it)